controller/arbitrage.js

```diff
@@ -196,6 +196,23 @@ class Arbitrage {
         }
     }
 
+    async getArbitrageTxData(txHash) {
+        const receipt = await C.web3.eth.getTransactionReceipt(txHash);
+        console.log(receipt);
+
+        if (receipt && receipt.logs) {
```

Comment:
I think it should just throw an error if there's no receipt or no logs, because otherwise it will return undefined and the calling code will fail.

Reply:
it was already like this, but yes, need to add a check in `calculateProfit`, good point

Reply:
ah, it is already in place, actually. It is inside a try catch block. All good then